pages/shared/[folderId].tsx

```diff
@@ -0,0 +1,73 @@
+import { useGetFolder } from "@/src/folder/data-access-folder";
+import { Layout } from "@/src/sharing/feature-layout";
+import { SharedLayout } from "@/src/page-layout/SharedLayout";
+import { CardList } from "@/src/link/ui-card-list";
+import { FolderInfo } from "@/src/folder/ui-folder-info";
+import { ReadOnlyCard } from "@/src/link/ui-read-only-card";
+import { SearchBar } from "@/src/link/ui-search-bar";
+import { useSearchLink } from "@/src/link/util-search-link/useSearchLink";
+import axios from "axios";
+import { useEffect, useState } from "react";
+
+interface UserData {
+  id: number;
+  name: string;
+  email: string;
+  image_source: string;
+}
+
+const SharedPage = () => {
+  const { data } = useGetFolder();
+  const { profileImage, ownerName, folderName, links } = data || {};
+  const { searchValue, handleChange, handleCloseClick, result } =
+    useSearchLink(links);
+
+  const [userData, setUserData] = useState<UserData | null>(null);
+
+  useEffect(() => {
+    const fetchUserData = async () => {
+      try {
+        const res = await axios.get<UserData>(
+          `https://bootcamp-api.codeit.kr/api/linkbrary/v1/sample/users`
+        );
+        setUserData(res.data);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    fetchUserData();
+  }, []);
+
+  return (
+    <Layout>
+      <SharedLayout
+        folderInfo={
+          userData && (
+            <FolderInfo
+              profileImage={userData.image_source}
+              ownerName={userData.name}
+              folderName={folderName}
+            />
+          )
+        }
```

Comment:
위 코드에서 userData는 배열입니다. 배열에 대해 `userData.name` 처럼 객체의 요소에 접근하는 방식을 사용하면 에러가 발생해요.
아래 코드처럼 배열의 map 메서드를 사용해서 userList 의 요소들 user 각각에 대한 FolderInfo 컴포넌트들을 렌더링하는 게 적절해 보입니다. 그런데 실제 화면 요구사항에 맞는 건지는 잘 모르겠네요 ㅎㅎ 일단 API 응답값과 현재 코드만 살펴봤을 때, 이렇게 할 수 있을 것 같아 제안드립니다.
```suggestion
        folderInfo={userList?.map(user => (
	        <FolderInfo
		        key={user.id}
		        profileImage={user.image_source}
		        ownerName={user.name}
		        folderName={folderName}
	        />
        ))}
```